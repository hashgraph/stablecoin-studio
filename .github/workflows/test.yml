name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test