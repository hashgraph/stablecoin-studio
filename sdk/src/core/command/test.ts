/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
