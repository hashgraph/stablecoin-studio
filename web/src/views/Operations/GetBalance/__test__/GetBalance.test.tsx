import { render } from '../../../../test/index';
import translations from '../../../../translations/en/getBalance.json';
import GetBalanceOperation from '../';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@testing-library/react';

describe(`<${GetBalanceOperation.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<GetBalanceOperation />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should render titles', () => {
		const component = render(<GetBalanceOperation />);

		expect(component.getByTestId('title')).toHaveTextContent(translations.title);
		expect(component.getByTestId('operation-title')).toHaveTextContent(translations.operationTitle);
	});

	test('should have an input to write the account', () => {
		const component = render(<GetBalanceOperation />);

		expect(component.getByTestId('account')).toBeInTheDocument();
	});

	test('should has an disabled confirm button that is enable when introduce an account', async () => {
		const component = render(<GetBalanceOperation />);

		const button = component.getByTestId('confirm-btn');
		expect(button).toBeDisabled();

		const account = component.getByTestId('account');
		userEvent.type(account, '0.0.12345');

		await waitFor(() => {
			expect(button).toBeEnabled();
		});
	});
});
