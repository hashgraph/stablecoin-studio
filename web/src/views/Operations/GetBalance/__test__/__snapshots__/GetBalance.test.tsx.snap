// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetBalanceOperation /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Operations
    </h2>
    <div
      class="css-1b6lda7"
      data-testid="base-container-children"
    >
      <div
        class="css-1us6gcn"
      >
        <div
          class="css-1r9sqqd"
        >
          <div
            class="chakra-stack css-n21gh5"
          >
            <h2
              class="chakra-heading css-17ipgvs"
              data-testid="title"
            >
              Get balance operation
            </h2>
            <p
              class="chakra-text css-1lyqlf8"
              data-testid="operation-title"
            >
              Operation details
            </p>
            <form
              class="chakra-stack css-hdflcv"
            >
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1hfoxt1"
                    for="field-:r1:"
                    id="field-:r1:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        What account?
                      </p>
                      <p
                        class="chakra-text css-ftb7fv"
                        data-testid="required"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-l1kxkf"
                      data-testid="targetAccount"
                      id="field-:r1:"
                      name="targetAccount"
                      placeholder="Introduce account"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            class="chakra-stack css-eunu1i"
          >
            <div
              class="chakra-stack css-wqnj46"
            >
              <h2
                class="chakra-heading css-m8uvwu"
                data-testid="details-title"
              >
                Stable coin details
              </h2>
              <div
                class="css-13brihr"
              >
                <p
                  class="chakra-text css-53sllx"
                  data-testid="details-review-title"
                >
                  Basic details
                </p>
                <div
                  class="css-0"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-0"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Name
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          ManuCoin2
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-1"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Symbol
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          MCC2
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-2"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Decimals
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          3
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-13brihr"
              >
                <p
                  class="chakra-text css-53sllx"
                  data-testid="details-review-title"
                >
                  Optional details
                </p>
                <div
                  class="css-0"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-0"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Initial supply
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          0
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-1"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Total supply
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          100
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-2"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Supply type
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          Infinite
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-3gu2fl"
        >
          <div
            class="chakra-stack css-1amf41o"
          >
            <button
              class="chakra-button css-1nrhi9n"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-1178n6s"
              data-testid="confirm-btn"
              disabled=""
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
